src/api/liskApi.js

```diff
@@ -13,211 +13,300 @@
-		this.nethash = this.getNethash(options.nethash);
+				: getDefaultPort(options);
+		this.randomizeNodes = options.randomizeNodes !== undefined
+			? options.randomizeNodes
+			: true;
+		this.providedNode = options.node || null;
+		this.node = options.node || this.selectNewNode();
 	}
 
-	getNethash(providedNethash) {
-		const { port } = this;
-		const NetHash = this.testnet
-			? utils.netHashOptions({ port }).testnet
-			: utils.netHashOptions({ port }).mainnet;
+	get allNodes() {
+		return {
+			current: this.node,
+			default: this.defaultNodes,
+			ssl: this.defaultSSLNodes,
+			testnet: this.defaultTestnetNodes,
+		};
+	}
 
-		if (providedNethash) {
-			NetHash.nethash = providedNethash;
-			NetHash.version = '0.0.0a';
+	get nodes() {
+		if (this.testnet) {
+			return this.defaultTestnetNodes;
+		}
+		if (this.ssl) {
+			return this.defaultSSLNodes;
+		}
+		return this.defaultNodes;
+	}
+
+	get randomNode() {
+		const nodes = this.nodes.filter(node => !this.isBanned(node));
+
+		if (!nodes.length || nodes.length === 0) {
+			throw new Error(
+				'Cannot get random node: all relevant nodes have been banned.',
+			);
 		}
 
-		return NetHash;
+		const randomIndex = Math.floor(Math.random() * nodes.length);
+		return nodes[randomIndex];
 	}
 
-	getNodes() {
-		return {
-			official: this.defaultNodes.map(node => ({ node })),
-			ssl: this.defaultSSLNodes.map(node => ({ node, ssl: true })),
-			testnet: this.defaultTestnetNodes.map(node => ({
-				node,
-				testnet: true,
-			})),
-		};
+	get urlPrefix() {
+		if (this.ssl) {
+			return 'https';
+		}
+		return 'http';
 	}
 
-	setNode(node) {
-		this.node = node || privateApi.selectNewNode.call(this);
-		return this.node;
+	get fullURL() {
+		const nodeUrl = this.port ? `${this.node}:${this.port}` : this.node;
+		return `${this.urlPrefix}://${nodeUrl}`;
 	}
 
-	setTestnet(testnet) {
-		if (this.testnet !== testnet) {
+	setSSL(newSSLValue) {
+		if (this.ssl !== newSSLValue) {
+			const nonSSLPort = this.testnet ? TEST_PORT : LIVE_PORT;
+
+			this.ssl = newSSLValue;
+			this.port = newSSLValue ? SSL_PORT : nonSSLPort;
 			this.bannedNodes = [];
-		}
-		this.testnet = testnet;
-		this.port = testnet ? TEST_PORT : LIVE_PORT;
 
-		privateApi.selectNewNode.call(this);
+			this.selectNewNode();
+		}
 	}
 
-	setSSL(ssl) {
-		if (this.ssl !== ssl) {
-			this.ssl = ssl;
+	setTestnet(newTestnetValue) {
+		if (this.testnet !== newTestnetValue) {
+			const nonTestnetPort = this.ssl ? SSL_PORT : LIVE_PORT;
+
+			this.testnet = newTestnetValue;
+			this.port = newTestnetValue ? TEST_PORT : nonTestnetPort;
 			this.bannedNodes = [];
-			privateApi.selectNewNode.call(this);
+
+			this.selectNewNode();
 		}
 	}
 
-	broadcastTransactions(transactions) {
-		return privateApi.sendRequestPromise
-			.call(this, POST, 'transactions', transactions)
-			.then(result => result.body);
+	banActiveNode() {
+		if (!this.isBanned(this.node)) {
+			this.bannedNodes.push(this.node);
+			this.node = this.selectNewNode();
```

Comment:
Selecting new node is ambiguous side effect.

Reply:
Might be solvable by renaming?